src/buddy/sign/jwt.clj

```diff
@@ -117,11 +117,11 @@
 
 (defn unsign
   ([message pkey] (unsign message pkey {}))
-  ([message pkey {:keys [skip-validation] :or {skip-validation false} :as opts}]
+  ([message pkey {:keys [skip-validation key-fn] :or {skip-validation false key-fn true} :as opts}]
    (try
      (let [claims (-> (jws/unsign message pkey opts)
                       (codecs/bytes->str)
-                      (json/parse-string true))]
+                      (json/parse-string key-fn))]
```

Comment:
I'd suggest wrapping `key-fn`, when explicitly given, inside a function that injects an intercepting conversion for the [registered claim names](https://tools.ietf.org/html/rfc7519#section-4.1). These claim names need to be keywords for the claim validation to work properly.